### PR TITLE
feat: add more default cluster roles

### LINKED ISSUE #90: Include default roles for Kubernetes
Currently, Infra requires users to have Kubernetes `roles` set up ahead of time. Infra simply takes care of mapping.

However, many users don't have Kubernetes roles set up ahead of time, and they have asked us to include "good default" roles as part of Infra:

* `readonly` Read (almost) all resources
* `edit` Edit (almost) all resources
* `port-forward` Allow port-forwarding
* `exec` Allow executing shell commands in pods
* `admin` Full access

Others:
* `rolling-restart` Allow deploying and restarting services
* `secrets` Allow secret access

### PR DESCRIPTION
**Pull request by mxyng**
## Summary

Filter ClusterRoles based on labels
- kubernetes.io/bootstrapping=rbac-defaults
- app.infrahq.com/include-role=true

## Checklist

- [ ] Wrote appropriate unit tests
- [ ] Considered security implications of the change
- [ ] Updated associated docs where necessary
- [ ] Updated associated configuration where necessary
- [ ] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [ ] Nothing sensitive logged
- [ ] Considered data migrations for smooth upgrades

## Related Issues

Resolves #90

### CODE REVIEW

File: helm/charts/infra/templates/connector/clusterrole.yaml [hunk 1]
```diff
@@ -36,4 +36,61 @@ rules:
...
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: pod-logs
+  labels:
+{{- include "connector.clusterRoleLabels" . | nindent 4 }}
+rules:
+  - apiGroups: [""]
+    resources:
+      - pods
```
Comment (mxyng) [inline helm/charts/infra/templates/connector/clusterrole.yaml:49]: This, and both cluster roles below, are pretty restrictive. `kubectl logs` also takes deployment, replica set, stateful set, daemon set, as arguments which this cluster role does not permit. To allow those use cases, we'll need to add `deployments`, `replicasets`, etc. to the cluster role resources

File: internal/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -596,13 +596,22 @@ func (k *Kubernetes) ClusterRoles() ([]string, error) {
 		return nil, err
 	}
 
-	clusterRoles, err := clientset.RbacV1().ClusterRoles().List(context.Background(), metav1.ListOptions{LabelSelector: "kubernetes.io/bootstrapping=rbac-defaults"})
+	rbacDefaults, err := clientset.RbacV1().ClusterRoles().List(context.Background(), metav1.ListOptions{
```
Comment (jmorganca) [inline internal/kubernetes/kubernetes.go:599]: Is there a way to combine these into one query?

File: helm/charts/infra/templates/connector/clusterrole.yaml (hunk 1 above)
Comment (jmorganca) [reply helm/charts/infra/templates/connector/clusterrole.yaml:49]: This is super cool. To enable users have `kubectl logs [deployment|rs|pod]` under one `logs` role, let's just do that here. Is there a downside to that?

File: helm/charts/infra/templates/connector/clusterrole.yaml [hunk 3]
```diff
@@ -36,4 +36,61 @@ rules:
       - watch
       - bind
       - escalate
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: pod-logs
```
Comment (jmorganca) [inline helm/charts/infra/templates/connector/clusterrole.yaml:43]: Re below comment: we can call this `logs`

File: internal/kubernetes/kubernetes.go (hunk 2 above)
Comment (mxyng) [reply internal/kubernetes/kubernetes.go:599]: Not that I'm aware. The value is taken as a union and I didn't see an OR operator

File: helm/charts/infra/templates/connector/clusterrole.yaml (hunk 1 above)
Comment (mxyng) [reply helm/charts/infra/templates/connector/clusterrole.yaml:49]: Not really. It gives more permissions but covers a broader set of use cases

File: helm/charts/infra/templates/connector/_helpers.tpl [hunk 4]
```diff
@@ -177,3 +177,11 @@ Infer whether Infra connector should be deployed based on connector.enabled, con
 {{- define "connector.enabled" -}}
 {{- or .Values.connector.enabled (not (empty .Values.connector.config.server)) (not (empty .Values.connector.config.accessKey)) }}
 {{- end }}
+
+{{/*
+ClusteRole labels
```
Comment (BruceMacD) [inline helm/charts/infra/templates/connector/_helpers.tpl:182]:
```suggestion
ClusterRole labels
```

Comment (BruceMacD) [reply helm/charts/infra/templates/connector/_helpers.tpl:182]: Should we include some more explanation here, like:
> Default ClusterRole labels created by Infra

**Review by BruceMacD — APPROVED** [review]

**Pull request merged**